frontend/src/metabase/notifications/modals/CreateOrEditQuestionAlertModal/CreateOrEditQuestionAlertModal.unit.spec.tsx

```diff
@@ -32,46 +33,54 @@ describe("CreateOrEditQuestionAlertModal", () => {
     expect(screen.getByText("Email")).toBeInTheDocument();
   });
 
-  it("should display first available channel by default - Slack", async () => {
-    setup({
-      isEmailSetup: false,
-      isSlackSetup: true,
-      isAdmin: true,
-    });
+  it.each([{ isAdmin: true }, { isAdmin: false, userCanAccessSettings: true }])(
+    "should display first available channel by default - Slack %p",
+    async setupConfig => {
+      setup({
+        isEmailSetup: false,
+        isSlackSetup: true,
+        ...setupConfig,
+      });
 
-    await waitFor(() => {
-      expect(screen.getByText("New alert")).toBeInTheDocument();
-    });
+      await waitFor(() => {
+        expect(screen.getByText("New alert")).toBeInTheDocument();
+      });
 
-    expect(screen.queryByText("Email")).not.toBeInTheDocument();
-    expect(screen.getByText("Slack")).toBeInTheDocument();
-  });
+      expect(screen.queryByText("Email")).not.toBeInTheDocument();
+      expect(screen.getByText("Slack")).toBeInTheDocument();
+    },
+  );
 
-  it("should display first available channel by default - Webhook", async () => {
-    const mockWebhook = createMockChannel();
-    setup({
-      isEmailSetup: false,
-      isHttpSetup: true,
-      isAdmin: true,
-      webhooksResult: [mockWebhook],
-    });
+  it.each([{ isAdmin: true }, { isAdmin: false, userCanAccessSettings: true }])(
+    "should display first available channel by default - Webhook %p",
+    async () => {
+      const mockWebhook = createMockChannel();
+      setup({
+        isEmailSetup: false,
+        isHttpSetup: true,
+        isAdmin: true,
+        webhooksResult: [mockWebhook],
```

Comment:
I believe you forgot to spread `setupConfig` here. feel free to resolve if it's intended.

Reply:
You are right, thanks. Now I need to debug why test is passing though 😃

Reply:
Fixed, thanks